ui/pages/confirmations/components/confirm/info/shared/advanced-details/advanced-details.tsx

```diff
@@ -0,0 +1,174 @@
+import { useCopyToClipboard } from '../../../../../../../hooks/useCopyToClipboard';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import {
+  currentConfirmationSelector,
+  getCustomNonceValue,
+  getNextSuggestedNonce,
+  getUseNonceField,
+} from '../../../../../../../selectors';
+import {
+  getNextNonce,
+  showModal,
+  updateCustomNonce,
+} from '../../../../../../../store/actions';
+import { useKnownMethodDataInTransaction } from '../../hooks/known-method-data-in-transaction';
+
+const NonceDetails = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getNextNonce());
+  }, [dispatch]);
+
+  const useNonceField = useSelector(getUseNonceField);
+  const nextNonce = useSelector(getNextSuggestedNonce);
+  const customNonceValue = useSelector(getCustomNonceValue);
+
+  const openEditNonceModal = () =>
+    dispatch(
+      showModal({
+        name: 'CUSTOMIZE_NONCE',
+        useNonceField,
+        nextNonce,
+        customNonceValue,
+        updateCustomNonce: (value: string) => {
+          dispatch(updateCustomNonce(value));
+        },
+        getNextNonce,
+      }),
+    );
+
+  const displayedNonce = customNonceValue || nextNonce;
+
+  return (
+    <Box padding={2}>
+      <ConfirmInfoRow
+        label={t('advancedDetailsNonceDesc')}
+        tooltip={t('advancedDetailsNonceTooltip')}
+      >
+        <ConfirmInfoRowText
+          text={`${displayedNonce}`}
+          onEditCallback={() => openEditNonceModal()}
+        />
+      </ConfirmInfoRow>
+    </Box>
+  );
+};
+
+const Divider = () => (
+  <Box
+    borderColor={BorderColor.borderMuted}
+    borderWidth={1}
+    width={BlockSize.Full}
+  />
+);
+
+const DataDetails = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { knownMethodData } =
+    useKnownMethodDataInTransaction(currentConfirmation);
+
+  if (!knownMethodData?.name) {
+    return null;
+  }
+
+  const getMethodDataString = (rawMethodData: {
+    name: string;
+    params: { type: string }[];
+  }): string => {
+    const paramsStr = rawMethodData.params.map((param) => param.type).join(',');
+    const result = `FUNCTION TYPE: ${rawMethodData.name} (${paramsStr})`;
+
+    return result;
+  };
+
+  const methodDataString = getMethodDataString(knownMethodData);
+
+  return (
+    <>
+      <Divider />
+      <Box padding={2}>
+        <ConfirmInfoRow label={t('advancedDetailsDataDesc')}>
+          <ConfirmInfoRowText text={methodDataString} />
+        </ConfirmInfoRow>
+      </Box>
+    </>
+  );
+};
+
+const HexDetails = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const [copied, handleCopy] = useCopyToClipboard();
+
+  if (!currentConfirmation?.txParams?.data) {
```

Comment:
I'm not sure of the normalisation up to this point, but could the `data` also be `0x` in which case we also wouldn't need to show it?

Reply:
I can't think of a scenario in which a contract interaction would have `0x` as data, because that would mean no function call is triggered, in which case the transaction would probably have a different type.

Reply:
Maybe one to consider in future if we re-use this component for transfers.